packages/calcite-components/src/components/text-area/text-area.tsx

```diff
@@ -426,34 +430,50 @@ export class TextArea
     };
   }
 
-  private replacePlaceholdersInMessages(): string {
+  private replacePlaceholdersInLongMessages(): string {
     return this.messages.tooLong
-      .replace("{maxLength}", this.localizedCharacterLengthObj.maxLength)
-      .replace("{currentLength}", this.localizedCharacterLengthObj.currentLength);
+      .replace("{currentLength}", this.localizedCharacterLengthObj.currentLength)
+      .replace("{maxLength}", this.localizedCharacterLengthObj.maxLength);
   }
 
-  private isCharacterLimitExceeded(): boolean {
+  private replacePlaceholdersInShortMessages(): string {
+    return this.messages.tooShort
+      .replace("{currentLength}", this.localizedCharacterLengthObj.currentLength)
+      .replace("{minLength}", this.localizedCharacterLengthObj.minLength);
+  }
+
+  private isCharacterOverMaxLimit(): boolean {
     return this.value?.length > this.maxLength;
   }
 
+  private isCharacterUnderMinLimit(): boolean {
+    return this.value?.length < this.minLength;
+  }
+
   // #endregion
 
   // #region Rendering
 
   override render(): JsxNode {
-    const hasFooter = this.startSlotHasElements || this.endSlotHasElements || !!this.maxLength;
+    const hasFooter =
+      this.startSlotHasElements || this.endSlotHasElements || !!this.maxLength || !!this.minLength;
     return (
       <InteractiveContainer disabled={this.disabled}>
         <textarea
           aria-describedby={this.guid}
           aria-errormessage={IDS.validationMessage}
-          ariaInvalid={this.status === "invalid" || this.isCharacterLimitExceeded()}
+          ariaInvalid={
+            this.status === "invalid" ||
+            this.isCharacterOverMaxLimit() ||
+            this.isCharacterUnderMinLimit()
+          }
           ariaLabel={getLabelText(this)}
           autofocus={this.el.autofocus}
           class={{
             [CSS.textArea]: true,
             [CSS.readOnly]: this.readOnly,
-            [CSS.textAreaInvalid]: this.isCharacterLimitExceeded(),
+            [CSS.textAreaInvalid]:
+              this.isCharacterOverMaxLimit() || this.isCharacterUnderMinLimit(),
```

Comment:
Let's DRY this up by storing and reusing the results of `isCharacterOverMaxLimit()` and `isCharacterUnderMinLimit()` within this render method.